src/components/calcite-action-bar/utils.ts

```diff
@@ -0,0 +1,58 @@
+import { forceUpdate } from "@stencil/core";
+
+const actionHeight = 48;
+const groupMargin = 16;
+
+const getMaxActionCount = ({ height, groupCount }: { height: number; groupCount: number }): number => {
+  return Math.floor((height - groupCount * groupMargin) / actionHeight);
+};
+
+export const getOverflowCount = ({
+  actionCount,
+  height,
+  groupCount
+}: {
+  actionCount: number;
+  height: number;
+  groupCount: number;
+}): number => {
+  const maxActionsCount = getMaxActionCount({ height, groupCount });
+  return actionCount >= maxActionsCount ? actionCount - maxActionsCount + 2 : 0;
+};
+
+export const overflowActions = ({
+  actionGroups,
+  expanded,
+  overflowCount
+}: {
+  actionGroups: HTMLCalciteActionGroupElement[];
+  expanded: boolean;
+  overflowCount: number;
+}): void => {
+  let neededToSlot = overflowCount;
+  actionGroups.reverse().forEach((group) => {
+    const groupActions = Array.from(group.querySelectorAll("calcite-action")).reverse();
+
+    groupActions.forEach((groupAction) => {
+      groupAction.removeAttribute("slot");
```

Comment:
Also, with the `overflowActionsDisabled` option to disable automatically slotting these things, they can just turn it off.

Reply:
> I'm not sure how modifying a slot attribute is that different.

I see your point. The way I was looking at it, those props indicate component state as users interact with it. `slot` does not fall into this. Maybe this won't be an issue because of that (`slot` not updated by VDOM since it's not derived from state).

> I am totally open to other suggestions though.

I don't have an alternative off the top of my head. I just brought this up as a consideration for VDOM use cases.